Functions/FirebaseFunctions/FIRFunctions.m

```diff
@@ -95,30 +95,57 @@ + (void)load {
   return @[ internalProvider ];
 }
 
+// A table mapping FIRApp instances to FIRFunctions instances, used to cache FIRFunctions instances
+// on a per-app basis. Does not retain either its keys or its values, since FIRApps may be deleted
+// without notifying their respective FIRFunctions instances, and FIRFunctions instances may be
+// short- lived.
++ (NSMapTable<FIRApp *, FIRFunctions *> *)instanceCache {
```

Comment:
Yep, that's what I was mentioning in my reply to Paul above. I'm leaning toward instead retaining the FIRFunctions instance strongly and the FIRApp weakly, so if the FIRApp is deleted the FIRFunctions instance will be released but otherwise it'll be retained.